### PR TITLE
Build docker test errors

### PR DESCRIPTION
**Pull request by aleksandr-mokrov**
Added filtering of packages without version when exporting workspace (local packages are excluded).
Added docker build logging in case of a build error

### CODE REVIEW

File: openfl/interface/workspace.py [hunk 1]
```diff
@@ -397,10 +396,16 @@ def dockerize_(context, base_image, save):
...
-
-    except Exception as e:
+    except docker.errors.BuildError as e:
+        for log in e.build_log:
+            msg = log.get('stream')
+            if msg:
+                echo(msg)
         echo('Failed to build the image\n' + str(e) + '\n')
         sys.exit(1)
+    except Exception as e:
```
Comment (igor-davidyuk) [inline openfl/interface/workspace.py]: could you please remove this last exception so we see the error traceback (or also print the traceback from here)

**Review by igor-davidyuk — APPROVED** [review]

File: openfl/interface/workspace.py [hunk 2]
```diff
@@ -397,10 +396,16 @@ def dockerize_(context, base_image, save):
...
+        echo('Failed to build the image with unexpected error\n' + str(e) + '\n')
+        sys.exit(1)
     else:
         echo('The workspace image has been built successfully!')
```
Comment (dmitryagapov) [inline openfl/interface/workspace.py]: Maybe we should delete this `else` block and put it on the end of `try` block?

File: openfl/interface/workspace.py [hunk 3]
```diff
@@ -171,7 +170,7 @@ def export_():
     # Create Zip archive of directory
     make_archive(archive_name, archive_type, tmp_dir)
 
-    echo(f'Workspace exported to archive: {archive_file_name}')
+    echo(f'!Workspace exported to archive: {archive_file_name}')
```
Comment (itrushkin) [inline openfl/interface/workspace.py]: This change seems occasional.

File: openfl/utilities/utils.py [hunk 4]
```diff
@@ -196,3 +196,11 @@ def validate_file_hash(file_path, expected_hash, chunk_size=8192):
 
     if h.hexdigest() != expected_hash:
         raise SystemError('ZIP File hash doesn\'t match expected file hash.')
+
+
+def is_package_has_version(package: str) -> bool:
```
Comment (itrushkin) [inline openfl/utilities/utils.py]: Function name has a grammatical mistake: in an interrogative sentence, the verb "have" must be used with the auxiliary verbs "do" or "does", not "is". 
Some correct versions of naming:
 - `is_package_versioned` - this one I like the most
 - `package_has_version`
 - `does_package_have_version`

**Review by itrushkin — CHANGES_REQUESTED** [review]

File: openfl/utilities/utils.py (hunk 4 above)
Comment (dmitryagapov) [reply openfl/utilities/utils.py]: @itrushkin, I'm not argee with you. Generally all functions that checks smth and used in `if` statement started with `is_`.

File: openfl/interface/workspace.py (hunk 1 above)
Comment (aleksandr-mokrov) [reply openfl/interface/workspace.py]: removed

File: openfl/interface/workspace.py (hunk 2 above)
Comment (aleksandr-mokrov) [reply openfl/interface/workspace.py]: I put it after

File: openfl/interface/workspace.py (hunk 3 above)
Comment (aleksandr-mokrov) [reply openfl/interface/workspace.py]: fixed

File: openfl/utilities/utils.py (hunk 4 above)
Comment (aleksandr-mokrov) [reply openfl/utilities/utils.py]: renamed to is_package_versioned

**Review by itrushkin — APPROVED** [review]

**Review by dmitryagapov — APPROVED** [review]

**Comment by alexey-gruzdev** [comment]
ok to test

**Pull request merged**